### PR TITLE
Additional unit tests for JNA, Cipher, Random, Utils testing error inputs

### PR DESCRIPTION
**Pull request by geoffreyblake**
Add unit tests that exercise error paths for JNA, Cipher, Random, and Utils parts of the package. Raises coverage overall by a few percentage points.  These tests touch some of the error checking deep in the library as much as possible.

Tested on MacOS AMD64, RHEL8 x86 and RHEL8 aarch64.

### CODE REVIEW

File: src/test/java/org/apache/commons/crypto/cipher/AbstractCipherTest.java [hunk 1]
```diff
@@ -147,6 +153,75 @@ public void cryptoTest() throws Exception {
         }
     }
 
+    @Test
+    public void testNullTransform() throws Exception {
+        for (String transform : transformations) {
```
Comment (vanzin) [inline src/test/java/org/apache/commons/crypto/cipher/AbstractCipherTest.java:158]: `transform` is not used in the body, is the loop really needed?

(I guess not. Then you can also use `@Test(expected=RuntimeException.class)`.)

File: src/test/java/org/apache/commons/crypto/cipher/AbstractCipherTest.java [hunk 2]
```diff
@@ -147,6 +153,75 @@ public void cryptoTest() throws Exception {
...
+                final CryptoCipher cipher = getCipher(null);
+                Assert.fail("Expected RuntimeException via NullPointerException");
+            } catch (final RuntimeException e) {
+            }
+        }
+    }
+
+    @Test
+    public void testInvalidTransform() throws Exception {
+        for (String transform : transformations) {
```
Comment (vanzin) [inline src/test/java/org/apache/commons/crypto/cipher/AbstractCipherTest.java:169]: Same question.

File: src/test/java/org/apache/commons/crypto/cipher/AbstractCipherTest.java [hunk 3]
```diff
@@ -147,6 +153,75 @@ public void cryptoTest() throws Exception {
...
+                final byte[] invalidKey = { 0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06,
+                    0x07, 0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f, 0x11 };
+                cipher.init(OpenSsl.ENCRYPT_MODE, new SecretKeySpec(invalidKey, "AES"), new IvParameterSpec(IV));
+                Assert.fail("Expected InvalidKeyException");
+            } catch (final InvalidKeyException ike) {
+            }
+        }
+    }
+
+    //@Test(expected = InvalidAlgorithmParameterException.class, timeout = 120000)
```
Comment (vanzin) [inline src/test/java/org/apache/commons/crypto/cipher/AbstractCipherTest.java:194]: delete

File: src/test/java/org/apache/commons/crypto/cipher/AbstractCipherTest.java [hunk 4]
```diff
@@ -147,6 +153,75 @@ public void cryptoTest() throws Exception {
...
+            } catch (final InvalidAlgorithmParameterException iape) {
+            }
+        }
+    }
+
+    @Test
+    public void testInvalidIVClass() throws Exception {
+        for (String transform : transformations) {
+            try {
+            final CryptoCipher cipher = getCipher(transform);
```
Comment (vanzin) [inline src/test/java/org/apache/commons/crypto/cipher/AbstractCipherTest.java:215]: Funky indentation.

File: src/test/java/org/apache/commons/crypto/cipher/OpenSslCipherTest.java [hunk 5]
```diff
@@ -136,6 +137,15 @@ public void testInvalidIV() throws Exception {
         cipher.init(OpenSsl.ENCRYPT_MODE, KEY, new IvParameterSpec(invalidIV));
     }
 
+    @Test(expected = InvalidAlgorithmParameterException.class, timeout = 120000)
+    public void testInvalidIVClass() throws Exception {
+        final OpenSsl cipher = OpenSsl.getInstance("AES/CTR/NoPadding");
+        Assert.assertNotNull(cipher);
+
+        cipher.init(OpenSsl.ENCRYPT_MODE, KEY, new GCMParameterSpec(IV.length, IV));
+        Assert.fail("Should have caught an InvalidAlgorithmParameterException");
```
Comment (vanzin) [inline src/test/java/org/apache/commons/crypto/cipher/OpenSslCipherTest.java:146]: Unnecessary (`@Test` tag covers it).

File: src/test/java/org/apache/commons/crypto/utils/UtilsTest.java [hunk 6]
```diff
@@ -41,4 +42,13 @@ public void testSplitOmitEmptyLine() {
     public void testSplitNull() {
         Assert.assertEquals(Collections.<String> emptyList(), Utils.splitClassNames(null, ","));
     }
+    @Test
```
Comment (vanzin) [inline src/test/java/org/apache/commons/crypto/utils/UtilsTest.java:45]: nit: add blank line between methods

Comment (geoffreyblake) [reply src/test/java/org/apache/commons/crypto/utils/UtilsTest.java:47]: Fixed

File: src/test/java/org/apache/commons/crypto/cipher/OpenSslCipherTest.java (hunk 5 above)
Comment (geoffreyblake) [reply src/test/java/org/apache/commons/crypto/cipher/OpenSslCipherTest.java]: Fixed

File: src/test/java/org/apache/commons/crypto/cipher/AbstractCipherTest.java (hunk 4 above)
Comment (geoffreyblake) [reply src/test/java/org/apache/commons/crypto/cipher/AbstractCipherTest.java]: Fixed all the odd formatting, redundant code concerns.

File: src/test/java/org/apache/commons/crypto/cipher/AbstractCipherTest.java [hunk 7]
```diff
@@ -147,6 +153,62 @@ public void cryptoTest() throws Exception {
...
     }
 
+    @Test(expected = RuntimeException.class)
+    public void testNullTransform() throws Exception {
+        final CryptoCipher cipher = getCipher(null);
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void testInvalidTransform() throws Exception {
+        final CryptoCipher cipher = getCipher("AES/CBR/NoPadding/garbage/garbage");
```
Comment (vanzin) [inline src/test/java/org/apache/commons/crypto/cipher/AbstractCipherTest.java:163]:
```suggestion
        getCipher("AES/CBR/NoPadding/garbage/garbage");
```

**Comment by vanzin** [comment]
Hey I can commit my own suggestions! Hope you don't mind me doing that.

**Comment by geoffreyblake** [comment]
Not at all.  Saves me the time of making the changes and pushing.   It appears the checks are passing, the code coverage has gone up too, looks ready to merge.

**Comment by vanzin** [comment]
Merging.

**Pull request merged**